doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
+:ref:`contribute guidelines<modules_contributing>`. They should also have a
+Zephyr developer that is commited to maintain the module code base.
+
+Zephyr depends on several categories of modules, including but not limited to:
 
 - Debugger integration
 - Silicon vendor Hardware Abstraction Layers (HALs)
 - Cryptography libraries
 - File Systems
 - Inter-Process Communication (IPC) libraries
 
+This page summarizes a list of policies and best practices which aim at
+better organizing the workflow in Zephyr modules.
+
+Module Repositories
+*******************
+
+* All modules included in the default manifest shall be hosted in repositories
+  under the Zephyr Project Github organization.
+
+* Module repository names should follow the convention of using lowercase
+  letters and dashes instead of underscores. This rule will apply to all
+  new module repositories.
+
+  .. note::
+
+     It is not required to modify the names of existing module repositories
+     to make them comply with the above naming convention.
+
+* If the module has an external (upstream) project repository, the module
+  repository should preserve the upstream repository folder structure.
+
+* Modules should use "zephyr" as the default name for the repository main
+  branch. Branches for specific purposes, for example, a module branch for
+  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+
+  .. note::
+
+     It is **not** recommended in module repositories to have a 'master'
+     branch mirroring the master branch of the external repository.
+
+.. _modules_synchronization:
+
+Synchronizing with upstream
+===========================
+
+Requirements for allowed practices
+----------------------------------
+
+Any changes to module repositories, as a result of synchronizing with,
+upstream shall be brought in by pull requests. These pull requests
+shall be merg-able (i.e. with Rebase & Merge option, using Github UI).
+This ensures that a linear history is preserved in the module repositories.
+It also allows to easily review the incoming changea and run Zephyr CI,
+git lint, identity, and license checks directly on the set of changes
+that are to be brought into the module.
+
+.. note::
+
+     Force-pushing to a module's main branch is not allowed.
+
+Allowed practices
+---------------------
+
+The following practices are allowed and should be followed in all modules.
+
+**Updating modules with a diff from upstream:**
+Upstream changes brought as a single commit (manual diff) in a PR against
+the module's main branch. This approach is simple and should be applicable
+to all modules with the downside of supressing the upstream history in the
+module repository.
+
+The commit message is expected to identify the upstream repository, the
+version to which the module is updated (upstream version, tag, commit,
+etc.), and the reason for the doing the update.
+
+**Cherry picking upstream commits:**
```

Comment:
We want to keep module source the same as upstream + our patches, that are needed for the module to work with zephyr, right? Because if we leave something out we may start to diverge and there will be problem. So, if we have to cherry pick everything at some point, from upstream to zephyr module, do we have to maintain the order of commits?

Reply:
Would be better to said Rebase uspsteram commit? Rebase keeps commit sequence.
Especially `rebase --onto` should be used here.

Reply:
Yes, but I don't want to add git terms at this point. The important is to stress that we cherry pick _all_ commits. i.e. all new content. Tthe way this is done does not matter here.

I will add a sentence to make this more clear.

Reply:
thansk @ioannisg